Until now I had some partial success with the following changes:

```diff
diff --git i/pkgs/by-name/co/codex/package.nix w/pkgs/by-name/co/codex/package.nix
index 2b9554534df4..6ded107a9ffd 100644
--- i/pkgs/by-name/co/codex/package.nix
+++ w/pkgs/by-name/co/codex/package.nix
@@ -4,16 +4,28 @@
   rustPlatform,
   fetchFromGitHub,
   installShellFiles,
+  boringssl,
+  clang,
   cmake,
-  git,
+  gitMinimal,
   makeBinaryWrapper,
   nix-update-script,
   pkg-config,
   openssl,
   ripgrep,
+  runCommand,
   versionCheckHook,
   installShellCompletions ? stdenv.buildPlatform.canExecute stdenv.hostPlatform,
 }:
+let
+  # boring-sys expects the static libraries in build/ instead of lib/
+  boringssl-wrapper = runCommand "boringssl-wrapper" { } ''
+    mkdir $out
+    cd $out
+    ln -s ${boringssl.out}/lib build
+    ln -s ${boringssl.dev}/include include
+  '';
+in
 rustPlatform.buildRustPackage (finalAttrs: {
   pname = "codex";
   version = "0.91.0";
@@ -30,15 +42,20 @@ rustPlatform.buildRustPackage (finalAttrs: {
   cargoHash = "sha256-FIjNADpaJdhr56mBa5sBac6uHnIsoJv8wQu3FdkP7yo=";

   nativeBuildInputs = [
+    clang
     cmake
-    git
+    gitMinimal
     installShellFiles
     makeBinaryWrapper
     pkg-config
+    rustPlatform.bindgenHook
   ];

   buildInputs = [ openssl ];

+  env.BORING_BSSL_PATH = "${boringssl-wrapper}";
+  env.LIBCLANG_PATH = "${lib.getLib clang.cc}/lib";
+
   # NOTE: part of the test suite requires access to networking, local shells,
   # apple system configuration, etc. since this is a very fast moving target
   # (for now), with releases happening every other day, constantly figuring out
```

Now I get

```
 > nix log /nix/store/5j4waxln9mkvn497z3df272c9j5fwb0b-codex-0.91.0.drv
Using versionCheckHook
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/m82pgz3snly9ra15s7w3zqkx11isddql-source
source root is source/codex-rs
Executing cargoSetupPostUnpackHook
Finished cargoSetupPostUnpackHook
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Executing cargoSetupPostPatchHook
Validating consistency between /build/source/codex-rs/Cargo.lock and /build/codex-0.91.0-vendor/Cargo.lock
Finished cargoSetupPostPatchHook
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing cargoBuildHook
cargoBuildHook flags: -j 16 --target x86_64-unknown-linux-gnu --offline --profile release
   Compiling proc-macro2 v1.0.95
   Compiling unicode-ident v1.0.18
   Compiling libc v0.2.177
   Compiling serde_core v1.0.228
   Compiling cfg-if v1.0.1
   Compiling serde v1.0.228
error (ignored): error: interrupted by the user

~/Dev/per/nixpkgs  - (vnvvxlz) via ❄️  impure (nix-shell-env) took 5s
 > nix log /nix/store/5j4waxln9mkvn497z3df272c9j5fwb0b-codex-0.91.0.drv > cat

~/Dev/per/nixpkgs  - (vnvvxlz) via ❄️  impure (nix-shell-env)
 > nix log /nix/store/5j4waxln9mkvn497z3df272c9j5fwb0b-codex-0.91.0.drv | cat
Using versionCheckHook
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/m82pgz3snly9ra15s7w3zqkx11isddql-source
source root is source/codex-rs
Executing cargoSetupPostUnpackHook
Finished cargoSetupPostUnpackHook
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Executing cargoSetupPostPatchHook
Validating consistency between /build/source/codex-rs/Cargo.lock and /build/codex-0.91.0-vendor/Cargo.lock
Finished cargoSetupPostPatchHook
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing cargoBuildHook
cargoBuildHook flags: -j 16 --target x86_64-unknown-linux-gnu --offline --profile release
   Compiling proc-macro2 v1.0.95
   Compiling unicode-ident v1.0.18
   Compiling libc v0.2.177
   Compiling serde_core v1.0.228
   Compiling cfg-if v1.0.1
   Compiling serde v1.0.228
   Compiling memchr v2.7.5
   Compiling autocfg v1.5.0
   Compiling log v0.4.29
   Compiling pin-project-lite v0.2.16
   Compiling portable-atomic v1.11.1
   Compiling critical-section v1.2.0
   Compiling allocator-api2 v0.2.21
   Compiling equivalent v1.0.2
   Compiling parking_lot_core v0.9.11
   Compiling itoa v1.0.15
   Compiling foldhash v0.2.0
   Compiling bitflags v2.10.0
   Compiling scopeguard v1.2.0
   Compiling futures-core v0.3.31
   Compiling bytes v1.10.1
   Compiling shlex v1.3.0
   Compiling find-msvc-tools v0.1.7
   Compiling ryu v1.0.20
   Compiling getrandom v0.3.3
   Compiling thiserror v2.0.17
   Compiling serde_json v1.0.145
   Compiling futures-io v0.3.31
   Compiling lock_api v0.4.13
   Compiling version_check v0.9.5
   Compiling regex-syntax v0.8.5
   Compiling hashbrown v0.16.0
   Compiling futures-sink v0.3.31
   Compiling slab v0.4.11
   Compiling zerocopy v0.8.26
   Compiling stable_deref_trait v1.2.0
   Compiling aho-corasick v1.1.3
   Compiling futures-channel v0.3.31
   Compiling futures-task v0.3.31
   Compiling pin-utils v0.1.0
   Compiling writeable v0.6.2
   Compiling percent-encoding v2.3.1
   Compiling quote v1.0.40
   Compiling litemap v0.8.0
   Compiling subtle v2.6.1
   Compiling typenum v1.18.0
   Compiling syn v2.0.104
   Compiling generic-array v0.14.7
   Compiling num-traits v0.2.19
   Compiling rustix v1.0.8
   Compiling once_cell v1.21.3
   Compiling indexmap v2.12.0
   Compiling linux-raw-sys v0.9.4
   Compiling fnv v1.0.7
   Compiling tracing-core v0.1.35
   Compiling http v1.3.1
   Compiling icu_properties_data v2.1.1
   Compiling jobserver v0.1.34
   Compiling base64 v0.22.1
   Compiling icu_normalizer_data v2.1.1
   Compiling cc v1.2.52
   Compiling form_urlencoded v1.2.1
   Compiling pkg-config v0.3.32
   Compiling cpufeatures v0.2.17
   Compiling heck v0.5.0
   Compiling fastrand v2.3.0
   Compiling utf8_iter v1.0.4
   Compiling anyhow v1.0.100
   Compiling mio v1.0.4
   Compiling signal-hook-registry v1.4.5
   Compiling socket2 v0.6.1
   Compiling http-body v1.0.1
   Compiling sha1_smol v1.0.1
   Compiling mime v0.3.17
   Compiling regex-automata v0.4.13
   Compiling crypto-common v0.1.6
   Compiling block-buffer v0.10.4
   Compiling rand_core v0.9.3
   Compiling getrandom v0.2.16
   Compiling digest v0.10.7
   Compiling ident_case v1.0.1
   Compiling strsim v0.11.1
   Compiling atomic-waker v1.1.2
   Compiling ipnet v2.11.0
   Compiling http-body-util v0.1.3
   Compiling ppv-lite86 v0.2.21
   Compiling sync_wrapper v1.0.2
   Compiling crossbeam-utils v0.8.21
   Compiling httparse v1.10.1
   Compiling httpdate v1.0.3
   Compiling try-lock v0.2.5
   Compiling unicase v2.8.1
   Compiling want v0.3.1
   Compiling tower-service v0.3.3
   Compiling zeroize v1.8.2
   Compiling rand_chacha v0.9.0
   Compiling winnow v0.7.13
   Compiling mime_guess v2.0.5
   Compiling rustls-pki-types v1.12.0
   Compiling rand v0.9.2
   Compiling ring v0.17.14
   Compiling sha1 v0.10.6
   Compiling toml_datetime v0.7.5+spec-1.1.0
   Compiling unicode-xid v0.2.6
   Compiling simd-adler32 v0.3.7
   Compiling adler2 v2.0.1
   Compiling untrusted v0.9.0
   Compiling socket2 v0.5.10
   Compiling either v1.15.0
   Compiling hex v0.4.3
   Compiling miniz_oxide v0.8.9
   Compiling openssl-probe v0.1.6
   Compiling vcpkg v0.2.15
   Compiling rustls v0.23.29
   Compiling tower-layer v0.3.3
   Compiling ahash v0.8.12
   Compiling utf8parse v0.2.2
   Compiling crc32fast v1.5.0
   Compiling openssl-sys v0.9.111
   Compiling anstyle v1.0.11
   Compiling synstructure v0.13.2
   Compiling serde_derive_internals v0.29.1
   Compiling toml_parser v1.0.6+spec-1.1.0
   Compiling darling_core v0.21.3
   Compiling rustls-native-certs v0.8.1
   Compiling foreign-types-shared v0.1.1
   Compiling rustversion v1.0.21
   Compiling openssl v0.10.73
   Compiling toml_edit v0.23.10+spec-1.0.0
   Compiling same-file v1.0.6
   Compiling dyn-clone v1.0.19
   Compiling walkdir v2.5.0
   Compiling foreign-types v0.3.2
   Compiling path-dedot v3.1.1
   Compiling thiserror v1.0.69
   Compiling native-tls v0.2.14
   Compiling static_assertions v1.1.0
   Compiling termcolor v1.4.1
   Compiling iri-string v0.7.8
   Compiling schemars v0.8.22
   Compiling proc-macro-crate v3.4.0
   Compiling regex v1.12.2
   Compiling path-absolutize v3.1.1
   Compiling flate2 v1.1.2
   Compiling anstyle-parse v0.2.7
   Compiling fdeflate v0.3.7
   Compiling pxfm v0.1.23
   Compiling num-integer v0.1.46
   Compiling icu_locale_data v2.1.1
   Compiling zune-core v0.5.0
   Compiling serde_derive v1.0.228
   Compiling tokio-macros v2.6.0
   Compiling tracing-attributes v0.1.31
   Compiling thiserror-impl v2.0.17
   Compiling zerofrom-derive v0.1.6
   Compiling yoke-derive v0.8.0
   Compiling futures-macro v0.3.31
   Compiling zerovec-derive v0.11.1
   Compiling displaydoc v0.2.5
   Compiling async-trait v0.1.89
   Compiling futures-util v0.3.31
   Compiling openssl-macros v0.1.1
   Compiling darling_macro v0.21.3
   Compiling tracing v0.1.43
   Compiling thiserror-impl v1.0.69
   Compiling ts-rs-macros v11.1.0
   Compiling zerofrom v0.1.6
   Compiling schemars_derive v0.8.22
   Compiling yoke v0.8.0
   Compiling colorchoice v1.0.4
   Compiling anstyle-query v1.1.3
   Compiling is_terminal_polyfill v1.70.1
   Compiling zune-jpeg v0.5.5
   Compiling tempfile v3.23.0
   Compiling anstream v0.6.19
   Compiling png v0.18.0
   Compiling zerovec v0.11.5
   Compiling zerotrie v0.2.2
   Compiling darling v0.21.3
   Compiling serde_with_macros v3.16.1
   Compiling moxcms v0.7.5
   Compiling sha2 v0.10.9
   Compiling option-ext v0.2.0
   Compiling byteorder-lite v0.1.0
   Compiling bytemuck v1.23.1
   Compiling icu_decimal_data v2.1.1
   Compiling parking v2.2.1
   Compiling tinystr v0.8.1
   Compiling potential_utf v0.1.4
   Compiling dirs-sys v0.5.0
   Compiling icu_collections v2.1.1
   Compiling icu_locale_core v2.1.1
   Compiling num-bigint v0.4.6
   Compiling serde_repr v0.1.20
   Compiling webpki-roots v1.0.2
   Compiling concurrent-queue v2.5.0
   Compiling serde_with v3.16.1
   Compiling lru v0.16.3
   Compiling encoding_rs v0.8.35
   Compiling syn v1.0.109
   Compiling dirs v6.0.0
   Compiling terminal_size v0.4.2
   Compiling clap_lex v0.7.5
   Compiling cfg_aliases v0.2.1
   Compiling iana-time-zone v0.1.63
   Compiling smallvec v1.15.1
   Compiling uuid v1.18.1
   Compiling futures-executor v0.3.31
   Compiling icu_provider v2.1.1
   Compiling parking_lot v0.12.4
   Compiling futures v0.3.31
   Compiling ts-rs v11.1.0
   Compiling nibble_vec v0.1.0
   Compiling fixed_decimal v0.7.0
   Compiling serde_urlencoded v0.7.1
   Compiling tokio v1.49.0
   Compiling icu_normalizer v2.1.1
   Compiling icu_properties v2.1.1
   Compiling icu_locale v2.1.1
   Compiling clap_builder v4.5.54
   Compiling codex-utils-absolute-path v0.91.0 (/build/source/codex-rs/utils/absolute-path)
   Compiling codex-git v0.91.0 (/build/source/codex-rs/utils/git)
   Compiling mcp-types v0.91.0 (/build/source/codex-rs/mcp-types)
   Compiling icu_decimal v2.1.1
   Compiling idna_adapter v1.2.1
   Compiling chrono v0.4.43
   Compiling idna v1.0.3
   Compiling strum_macros v0.27.2
   Compiling clap_derive v4.5.49
   Compiling rand_core v0.6.4
   Compiling strum v0.27.2
   Compiling sys-locale v0.3.2
   Compiling url v2.5.4
   Compiling rand_chacha v0.3.1
   Compiling event-listener v5.4.0
   Compiling unicode-width v0.1.14
   Compiling unicode-segmentation v1.12.0
   Compiling lazy_static v1.5.0
   Compiling rand v0.8.5
   Compiling image v0.25.9
   Compiling event-listener-strategy v0.5.4
   Compiling ref-cast v1.0.24
   Compiling futures-lite v2.6.1
   Compiling ref-cast-impl v1.0.24
   Compiling enumflags2_derive v0.7.12
   Compiling zvariant_utils v2.1.0
   Compiling dirs-sys-next v0.1.2
   Compiling rustls-webpki v0.103.4
   Compiling crunchy v0.2.4
   Compiling clap v4.5.54
   Compiling nu-ansi-term v0.50.1
   Compiling powerfmt v0.2.0
   Compiling num-conv v0.1.0
   Compiling time-core v0.1.6
   Compiling deranged v0.5.4
   Compiling time-macros v0.2.24
   Compiling enumflags2 v0.7.12
   Compiling dirs-next v2.0.0
   Compiling sharded-slab v0.1.7
   Compiling matchers v0.2.0
   Compiling block-padding v0.3.3
   Compiling tracing-log v0.2.0
   Compiling memoffset v0.9.1
   Compiling thread_local v1.1.9
   Compiling num_threads v0.1.7
   Compiling tiny-keccak v2.0.2
   Compiling siphasher v1.0.1
   Compiling regex-syntax v0.6.29
   Compiling async-task v4.7.1
   Compiling cfg_aliases v0.1.1
   Compiling paste v1.0.15
   Compiling nix v0.28.0
   Compiling tracing-subscriber v0.3.22
   Compiling phf_shared v0.11.3
   Compiling inout v0.1.4
   Compiling tokio-util v0.7.18
   Compiling tokio-native-tls v0.3.1
   Compiling codex-utils-cache v0.91.0 (/build/source/codex-rs/utils/cache)
   Compiling term v0.7.0
   Compiling codex-utils-image v0.91.0 (/build/source/codex-rs/utils/image)
   Compiling zvariant_derive v4.2.0
   Compiling async-channel v2.5.0
   Compiling nix v0.29.0
   Compiling codex-protocol v0.91.0 (/build/source/codex-rs/protocol)
   Compiling time v0.3.44
   Compiling dupe_derive v0.9.1
   Compiling itertools v0.13.0
   Compiling h2 v0.4.11
   Compiling tower v0.5.2
   Compiling tokio-stream v0.1.18
   Compiling crossbeam-epoch v0.9.18
   Compiling piper v0.2.4
   Compiling async-io v2.6.0
   Compiling precomputed-hash v0.1.1
   Compiling minimal-lexical v0.2.1
   Compiling endi v1.1.0
   Compiling windows_x86_64_gnu v0.52.6
   Compiling fixedbitset v0.4.2
   Compiling new_debug_unreachable v1.0.6
   Compiling bit-vec v0.6.3
   Compiling petgraph v0.6.5
   Compiling nom v7.1.3
   Compiling ena v0.14.3
   Compiling bit-set v0.5.3
   Compiling tower-http v0.6.6
   Compiling string_cache v0.8.9
   Compiling zvariant v4.2.0
   Compiling blocking v1.6.2
   Compiling ascii-canvas v3.0.0
   Compiling cipher v0.4.4
   Compiling async-lock v3.4.1
   Compiling lalrpop-util v0.19.12
   Compiling tokio-rustls v0.26.2
   Compiling opentelemetry v0.31.0
   Compiling polling v3.11.0
   Compiling itertools v0.10.5
   Compiling is-terminal v0.4.16
   Compiling allocative v0.3.4
   Compiling data-encoding v2.10.0
   Compiling diff v0.1.13
   Compiling async-fs v2.2.0
   Compiling Inflector v0.11.4
   Compiling async-executor v1.13.3
   Compiling convert_case v0.6.0
   Compiling lalrpop v0.19.12
   Compiling ctor v0.1.26
   Compiling zbus_macros v4.4.0
   Compiling async-broadcast v0.7.2
   Compiling num-rational v0.4.2
   Compiling schemafy_core v0.5.2
   Compiling nix v0.30.1
   Compiling zbus_names v3.0.0
   Compiling num-iter v0.1.45
   Compiling allocative_derive v0.3.3
   Compiling hashbrown v0.14.5
   Compiling darling_core v0.23.0
   Compiling hyper v1.7.0
   Compiling zstd-sys v2.0.16+zstd.1.5.7
   Compiling hmac v0.12.1
   Compiling xdg-home v1.3.0
   Compiling num-complex v0.4.6
   Compiling ordered-stream v0.2.0
   Compiling byteorder v1.5.0
   Compiling beef v0.5.2
   Compiling starlark_map v0.13.0
   Compiling downcast-rs v1.2.1
   Compiling indenter v0.3.3
   Compiling bumpalo v3.19.0
   Compiling fxhash v0.2.1
   Compiling logos-derive v0.12.1
   Compiling hkdf v0.12.4
   Compiling schemafy_lib v0.5.2
   Compiling num v0.4.3
   Compiling derive_more-impl v1.0.0
   Compiling windows-targets v0.52.6
   Compiling aes v0.8.4
   Compiling cbc v0.1.2
   Compiling dupe v0.9.1
   Compiling hyper-util v0.1.16
   Compiling memoffset v0.6.5
   Compiling endian-type v0.1.2
   Compiling bitflags v1.3.2
   Compiling chunked_transfer v1.5.0
   Compiling ascii v1.1.0
   Compiling lsp-types v0.94.1
   Compiling zbus v4.4.0
   Compiling radix_trie v0.2.1
   Compiling tiny_http v0.12.0
   Compiling logos v0.12.1
   Compiling darling_macro v0.23.0
   Compiling schemafy v0.5.2
   Compiling darling v0.23.0
   Compiling windows-result v0.2.0
   Compiling opentelemetry_sdk v0.31.0
   Compiling derivative v2.2.0
   Compiling annotate-snippets v0.9.2
   Compiling hyper-tls v0.6.0
   Compiling hyper-rustls v0.27.7
   Compiling reqwest v0.12.24
   Compiling toml v0.5.11
   Compiling fd-lock v4.0.4
   Compiling pin-project-internal v1.1.10
   Compiling schemars_derive v1.0.4
   Compiling bstr v1.12.0
   Compiling itertools v0.14.0
   Compiling serde_path_to_error v0.1.20
   Compiling derive_more v1.0.0
   Compiling linux-keyutils v0.2.4
   Compiling zstd-safe v7.2.4
   Compiling home v0.5.11
   Compiling starlark v0.13.0
   Compiling rustyline v14.0.0
   Compiling pin-project v1.1.10
   Compiling winres v0.1.12
   Compiling process-wrap v9.0.0
   Compiling hyper-timeout v0.5.2
   Compiling debugserver-types v0.5.0
   Compiling starlark_derive v0.13.0
   Compiling windows-strings v0.1.0
   Compiling rmcp-macros v0.12.0
   Compiling schemars v1.0.4
   Compiling oauth2 v5.0.0
   Compiling prost-derive v0.14.1
   Compiling display_container v0.9.0
   Compiling textwrap v0.11.0
   Compiling sse-stream v0.2.1
   Compiling erased-serde v0.3.31
   Compiling tree-sitter v0.25.10
   Compiling axum-core v0.5.2
   Compiling windows-interface v0.58.0
   Compiling secret-service v4.0.0
   Compiling windows-implement v0.58.0
   Compiling strsim v0.10.0
   Compiling indoc v2.0.6
   Compiling inventory v0.3.20
   Compiling cmp_any v0.8.1
   Compiling maplit v1.0.2
   Compiling matchit v0.8.4
   Compiling tree-sitter-language v0.1.5
   Compiling pastey v0.2.0
   Compiling starlark_syntax v0.13.0
   Compiling env_home v0.1.0
   Compiling which v8.0.0
   Compiling windows-core v0.58.0
   Compiling prost v0.14.1
   Compiling axum v0.8.4
   Compiling tonic v0.14.2
   Compiling rmcp v0.12.0
   Compiling codex-windows-sandbox v0.91.0 (/build/source/codex-rs/windows-sandbox-rs)
   Compiling codex-app-server-protocol v0.91.0 (/build/source/codex-rs/app-server-protocol)
   Compiling keyring v3.6.3
   Compiling codex-keyring-store v0.91.0 (/build/source/codex-rs/keyring-store)
   Compiling webbrowser v1.0.6
   Compiling multimap v0.10.1
   Compiling env_filter v0.1.3
   Compiling tree-sitter-bash v0.25.0
   Compiling regex-lite v0.1.8
   Compiling urlencoding v2.1.3
   Compiling streaming-iterator v0.1.9
   Compiling utf-8 v0.7.6
   Compiling jiff v0.2.15
   Compiling tungstenite v0.28.0 (https://github.com/JakkuSakura/tungstenite-rs?rev=f514de8644821113e5d18a027d6d28a5c8cc0a6e#f514de86)
   Compiling tonic-prost v0.14.2
   Compiling windows v0.58.0
   Compiling tracing-opentelemetry v0.32.0
   Compiling globset v0.4.16
   Compiling eventsource-stream v0.2.3
   Compiling crossbeam-deque v0.8.6
   Compiling const-hex v1.17.0
   Compiling toml_writer v1.0.6+spec-1.1.0
   Compiling dunce v1.0.5
   Compiling opentelemetry-proto v0.31.0
   Compiling ignore v0.4.23
   Compiling tokio-tungstenite v0.28.0 (https://github.com/JakkuSakura/tokio-tungstenite?rev=2ae536b0de793f3ddf31fc2f22d445bf1ef2023d#2ae536b0)
   Compiling env_logger v0.11.8
   Compiling opentelemetry-http v0.31.0
   Compiling nucleo-matcher v0.3.1
   Compiling serde_spanned v1.0.0
   Compiling filedescriptor v0.8.3
   Compiling test-case-core v3.3.1
   Compiling serial2 v0.2.31
   Compiling shell-words v1.1.0
   Compiling similar v2.7.0
   Compiling portable-pty v0.9.0
   Compiling codex-file-search v0.91.0 (/build/source/codex-rs/file-search)
   Compiling test-case-macros v3.3.1
   Compiling codex-apply-patch v0.91.0 (/build/source/codex-rs/apply-patch)
   Compiling opentelemetry-otlp v0.31.0
   Compiling codex-rmcp-client v0.91.0 (/build/source/codex-rs/rmcp-client)
   Compiling toml v0.9.5
   Compiling opentelemetry-appender-tracing v0.31.1
   Compiling test-log-macros v0.2.19
   Compiling include_dir_macros v0.7.4
   Compiling unsafe-libyaml v0.2.11
   Compiling opentelemetry-semantic-conventions v0.31.0
   Compiling include_dir v0.7.4
   Compiling test-log v0.2.19
   Compiling test-case v3.3.1
   Compiling codex-utils-pty v0.91.0 (/build/source/codex-rs/utils/pty)
   Compiling toml_edit v0.24.0+spec-1.1.0
   Compiling codex-utils-readiness v0.91.0 (/build/source/codex-rs/utils/readiness)
   Compiling serde_yaml v0.9.34+deprecated
   Compiling codex-async-utils v0.91.0 (/build/source/codex-rs/async-utils)
   Compiling landlock v0.4.4
   Compiling chardetng v0.1.17
   Compiling os_info v3.12.0
   Compiling arc-swap v1.8.0
   Compiling seccompiler v0.5.0
   Compiling codex-utils-string v0.91.0 (/build/source/codex-rs/utils/string)
   Compiling env-flags v0.1.1
   Compiling wildmatch v2.6.1
   Compiling async-stream-impl v0.3.6
   Compiling num_cpus v1.17.0
   Compiling deadpool-runtime v0.1.4
   Compiling deadpool v0.12.3
   Compiling assert-json-diff v2.0.2
   Compiling debugid v0.8.0
   Compiling rustix v0.38.44
   Compiling wiremock v0.6.5
   Compiling async-stream v0.3.6
   Compiling semver v1.0.27
   Compiling object v0.36.7
   Compiling sentry-types v0.46.0
   Compiling const_format_proc_macros v0.2.34
   Compiling crossbeam-channel v0.5.15
   Compiling linux-raw-sys v0.4.15
   Compiling gimli v0.31.1
   Compiling wildcard v0.3.0
   Compiling const_format v0.2.35
   Compiling rama-macros v0.3.0-alpha.4
   Compiling wayland-sys v0.31.7
   Compiling rustc-demangle v0.1.25
   Compiling dotenvy v0.15.7
   Compiling signal-hook v0.3.18
   Compiling smol_str v0.3.5
   Compiling tokio-graceful v0.2.2
   Compiling rustc_version v0.4.1
   Compiling darling_core v0.20.11
   Compiling rama-utils v0.3.0-alpha.4
   Compiling wayland-backend v0.3.11
   Compiling findshlibs v0.10.2
   Compiling asynk-strim v0.1.5
   Compiling addr2line v0.24.2
   Compiling predicates-core v1.0.9
   Compiling glob v0.3.3
   Compiling is_ci v1.2.0
   Compiling unicode-width v0.2.1
   Compiling rama-error v0.3.0-alpha.4
   Compiling doc-comment v0.3.3
   Compiling clang-sys v1.8.1
   Compiling rama-core v0.3.0-alpha.4
   Compiling sentry-contexts v0.46.0
   Compiling sentry-core v0.46.0
   Compiling quick-xml v0.37.5
   Compiling spin v0.9.8
   Compiling nom v8.0.0
   Compiling assert_cmd v2.0.17
   Compiling wayland-client v0.31.11
   Compiling owo-colors v4.2.2
   Compiling instability v0.3.9
   Compiling backtrace v0.3.75
   Compiling foldhash v0.1.5
   Compiling darling_macro v0.20.11
   Compiling termtree v0.5.1
   Compiling endian-type v0.2.0
   Compiling psl-types v2.0.11
   Compiling difflib v0.4.0
   Compiling psl v2.1.178
   Compiling darling v0.20.11
   Compiling predicates v3.1.3
   Compiling sentry-backtrace v0.46.0
   Compiling radix_trie v0.3.0
   Compiling predicates-tree v1.0.12
   Compiling hashbrown v0.15.4
   Compiling wayland-scanner v0.31.7
   Compiling flume v0.12.0
   Compiling signal-hook-mio v0.2.4
   Compiling supports-color v2.1.0
   Compiling supports-color v3.0.2
   Compiling castaway v0.2.4
   Compiling strum_macros v0.26.4
   Compiling uname v0.1.1
   Compiling hostname v0.4.1
   Compiling wait-timeout v0.2.1
   Compiling md5 v0.8.0
   Compiling lru v0.12.5
   Compiling sentry-debug-images v0.46.0
   Compiling rama-http-types v0.3.0-alpha.4
   Compiling compact_str v0.8.1
   Compiling crossterm v0.28.1 (https://github.com/nornagon/crossterm?branch=nornagon%2Fcolor-query#87db8bfa)
   Compiling sentry-panic v0.46.0
   Compiling unicode-truncate v1.1.0
   Compiling convert_case v0.10.0
   Compiling codex-backend-openapi-models v0.91.0 (/build/source/codex-rs/codex-backend-openapi-models)
   Compiling libloading v0.8.9
   Compiling bindgen v0.72.1
   Compiling cassowary v0.3.0
   Compiling strum v0.26.3
   Compiling cexpr v0.6.0
   Compiling derive_more-impl v2.1.1
   Compiling sentry v0.46.0
   Compiling ratatui v0.29.0 (https://github.com/nornagon/ratatui?branch=nornagon-v0.29.0-patch#9b2ad129)
   Compiling wayland-protocols v0.32.9
   Compiling codex-utils-cargo-bin v0.91.0 (/build/source/codex-rs/utils/cargo-bin)
   Compiling rama-net v0.3.0-alpha.4
   Compiling eyre v0.6.12
   Compiling color-spantrace v0.3.0
   Compiling rustc-hash v2.1.1
   Compiling derive_more v2.1.1
   Compiling tree_magic_mini v3.2.0
   Compiling codex-feedback v0.91.0 (/build/source/codex-rs/feedback)
   Compiling tracing-error v0.2.1
   Compiling cmake v0.1.57
   Compiling os_pipe v1.2.2
   Compiling fslock v0.2.1
   Compiling pulldown-cmark v0.10.3
   Compiling fs_extra v1.3.0
   Compiling codex-execpolicy v0.91.0 (/build/source/codex-rs/execpolicy)
   Compiling tinyvec_macros v0.1.1
   Compiling x11rb-protocol v0.13.1
   Compiling tagptr v0.2.0
   Compiling simdutf8 v0.1.5
   Compiling ansi-to-tui v7.0.0
   Compiling moka v0.12.12
   Compiling tinyvec v1.10.0
   Compiling wayland-protocols-wlr v0.3.9
   Compiling getopts v0.2.23
   Compiling codex-utils-json-to-toml v0.91.0 (/build/source/codex-rs/utils/json-to-toml)
   Compiling diffy v0.4.2
   Compiling enum-as-inner v0.6.1
   Compiling rama-http-headers v0.3.0-alpha.4
   Compiling unicode-linebreak v0.1.5
   Compiling wl-clipboard-rs v0.9.2
   Compiling smawk v0.3.2
   Compiling pulldown-cmark-escape v0.10.1
   Compiling textwrap v0.16.2
   Compiling hickory-proto v0.25.2
   Compiling color-eyre v0.6.5
   Compiling codex-ansi-escape v0.91.0 (/build/source/codex-rs/ansi-escape)
   Compiling x11rb v0.13.1
   Compiling ratatui-macros v0.6.0
   Compiling tracing-appender v0.2.3
   Compiling tree-sitter-highlight v0.25.10
   Compiling foreign-types-macros v0.2.3
   Compiling csv-core v0.1.13
   Compiling foreign-types-shared v0.3.1
   Compiling arboard v3.6.1
   Compiling pathdiff v0.2.3
   Compiling resolv-conf v0.7.6
   Compiling rama-boring-sys v0.5.9
   Compiling dtor-proc-macro v0.0.6
   Compiling foreign-types v0.5.0
   Compiling csv v1.4.0
   Compiling serde_html_form v0.3.2
   Compiling inotify-sys v0.1.5
   Compiling dtor v0.1.0
   Compiling http-range-header v0.4.2
   Compiling matchit v0.9.1
   Compiling ctor-proc-macro v0.0.7
   Compiling ctor v0.6.3
   Compiling inotify v0.11.0
   Compiling tokio-test v0.4.4
   Compiling codex-process-hardening v0.91.0 (/build/source/codex-rs/process-hardening)
   Compiling notify-types v2.0.0
   Compiling codex-responses-api-proxy v0.91.0 (/build/source/codex-rs/responses-api-proxy)
   Compiling rama-unix v0.3.0-alpha.4
   Compiling codex-stdio-to-uds v0.91.0 (/build/source/codex-rs/stdio-to-uds)
   Compiling notify v8.2.0
   Compiling codex-execpolicy-legacy v0.91.0 (/build/source/codex-rs/execpolicy-legacy)
   Compiling clap_complete v4.5.64
   Compiling yansi v1.0.1
   Compiling hickory-resolver v0.25.2
   Compiling pretty_assertions v1.4.1
   Compiling rama-dns v0.3.0-alpha.4
   Compiling rama-http v0.3.0-alpha.4
   Compiling rama-boring v0.5.9
   Compiling rama-tcp v0.3.0-alpha.4
error[E0425]: cannot find value `TLSEXT_TYPE_record_size_limit` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:543:51
    |
543 |     pub const RECORD_SIZE_LIMIT: Self = Self(ffi::TLSEXT_TYPE_record_size_limit as u16);
    |                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0425]: cannot find value `SSL_GROUP_X25519_KYBER512_DRAFT00` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:712:23
    |
712 |         SslCurve(ffi::SSL_GROUP_X25519_KYBER512_DRAFT00 as _);
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `SSL_GROUP_X25519_KYBER768_DRAFT00`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:228265
    |
  3 | ...88 ; pub const SSL_GROUP_X25519_KYBER768_DRAFT00 : i32 = 25497 ; pub const SSL_GROUP_MLKEM1024 : i32 = 514 ; pub const SSL_VERIFY_NONE...
    |         ------------------------------------------------- similarly named constant `SSL_GROUP_X25519_KYBER768_DRAFT00` defined here

error[E0425]: cannot find value `SSL_GROUP_X25519_KYBER768_DRAFT00_OLD` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:715:23
    |
715 |         SslCurve(ffi::SSL_GROUP_X25519_KYBER768_DRAFT00_OLD as _);
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `SSL_GROUP_X25519_KYBER768_DRAFT00`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:228265
    |
  3 | ...88 ; pub const SSL_GROUP_X25519_KYBER768_DRAFT00 : i32 = 25497 ; pub const SSL_GROUP_MLKEM1024 : i32 = 514 ; pub const SSL_VERIFY_NONE...
    |         ------------------------------------------------- similarly named constant `SSL_GROUP_X25519_KYBER768_DRAFT00` defined here

error[E0425]: cannot find value `SSL_GROUP_P256_KYBER768_DRAFT00` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:717:63
    |
717 |     pub const P256_KYBER768_DRAFT00: SslCurve = SslCurve(ffi::SSL_GROUP_P256_KYBER768_DRAFT00 as _);
    |                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `SSL_GROUP_X25519_KYBER768_DRAFT00`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:228265
    |
  3 | ...88 ; pub const SSL_GROUP_X25519_KYBER768_DRAFT00 : i32 = 25497 ; pub const SSL_GROUP_MLKEM1024 : i32 = 514 ; pub const SSL_VERIFY_NONE...
    |         ------------------------------------------------- similarly named constant `SSL_GROUP_X25519_KYBER768_DRAFT00` defined here

error[E0531]: cannot find unit struct, unit variant or constant `SSL_GROUP_X25519_KYBER512_DRAFT00` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:755:18
    |
755 |             ffi::SSL_GROUP_X25519_KYBER512_DRAFT00 => Some(ffi::NID_X25519Kyber512Draft00),
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `SSL_GROUP_X25519_KYBER768_DRAFT00`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:228265
    |
  3 | ...88 ; pub const SSL_GROUP_X25519_KYBER768_DRAFT00 : i32 = 25497 ; pub const SSL_GROUP_MLKEM1024 : i32 = 514 ; pub const SSL_VERIFY_NONE...
    |         ------------------------------------------------- similarly named constant `SSL_GROUP_X25519_KYBER768_DRAFT00` defined here

error[E0425]: cannot find value `NID_X25519Kyber512Draft00` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:755:65
    |
755 |             ffi::SSL_GROUP_X25519_KYBER512_DRAFT00 => Some(ffi::NID_X25519Kyber512Draft00),
    |                                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `NID_X25519Kyber768Draft00`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:174072
    |
  3 | ...Draft00\0" ; pub const NID_X25519Kyber768Draft00 : i32 = 964 ; pub const SN_X25519MLKEM768 : & [u8 ; 15] = b"X25519MLKEM768\0" ; pub c...
    |                 ----------------------------------------- similarly named constant `NID_X25519Kyber768Draft00` defined here

error[E0531]: cannot find unit struct, unit variant or constant `SSL_GROUP_X25519_KYBER768_DRAFT00_OLD` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:756:18
    |
756 |             ffi::SSL_GROUP_X25519_KYBER768_DRAFT00_OLD => Some(ffi::NID_X25519Kyber768Draft00Old),
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `SSL_GROUP_X25519_KYBER768_DRAFT00`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:228265
    |
  3 | ...88 ; pub const SSL_GROUP_X25519_KYBER768_DRAFT00 : i32 = 25497 ; pub const SSL_GROUP_MLKEM1024 : i32 = 514 ; pub const SSL_VERIFY_NONE...
    |         ------------------------------------------------- similarly named constant `SSL_GROUP_X25519_KYBER768_DRAFT00` defined here

error[E0425]: cannot find value `NID_X25519Kyber768Draft00Old` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:756:69
    |
756 |             ffi::SSL_GROUP_X25519_KYBER768_DRAFT00_OLD => Some(ffi::NID_X25519Kyber768Draft00Old),
    |                                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `NID_X25519Kyber768Draft00`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:174072
    |
  3 | ...Draft00\0" ; pub const NID_X25519Kyber768Draft00 : i32 = 964 ; pub const SN_X25519MLKEM768 : & [u8 ; 15] = b"X25519MLKEM768\0" ; pub c...
    |                 ----------------------------------------- similarly named constant `NID_X25519Kyber768Draft00` defined here

error[E0531]: cannot find unit struct, unit variant or constant `SSL_GROUP_P256_KYBER768_DRAFT00` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:757:18
    |
757 |             ffi::SSL_GROUP_P256_KYBER768_DRAFT00 => Some(ffi::NID_P256Kyber768Draft00),
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `SSL_GROUP_X25519_KYBER768_DRAFT00`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:228265
    |
  3 | ...88 ; pub const SSL_GROUP_X25519_KYBER768_DRAFT00 : i32 = 25497 ; pub const SSL_GROUP_MLKEM1024 : i32 = 514 ; pub const SSL_VERIFY_NONE...
    |         ------------------------------------------------- similarly named constant `SSL_GROUP_X25519_KYBER768_DRAFT00` defined here

error[E0425]: cannot find value `NID_P256Kyber768Draft00` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:757:63
    |
757 |             ffi::SSL_GROUP_P256_KYBER768_DRAFT00 => Some(ffi::NID_P256Kyber768Draft00),
    |                                                               ^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `NID_X25519Kyber768Draft00`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:174072
    |
  3 | ...Draft00\0" ; pub const NID_X25519Kyber768Draft00 : i32 = 964 ; pub const SN_X25519MLKEM768 : & [u8 ; 15] = b"X25519MLKEM768\0" ; pub c...
    |                 ----------------------------------------- similarly named constant `NID_X25519Kyber768Draft00` defined here

error[E0425]: cannot find value `NID_MLKEM1024` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:758:51
    |
758 |             ffi::SSL_GROUP_MLKEM1024 => Some(ffi::NID_MLKEM1024),
    |                                                   ^^^^^^^^^^^^^ help: a constant with a similar name exists: `NID_ML_KEM_1024`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:174358
    |
  3 | ...] = b"ML-KEM-1024\0" ; pub const NID_ML_KEM_1024 : i32 = 966 ; pub const SN_ML_DSA_44 : & [u8 ; 13] = b"id-ml-dsa-44\0" ; pub const LN...
    |                           ------------------------------- similarly named constant `NID_ML_KEM_1024` defined here

error[E0425]: cannot find value `TLSEXT_cert_compression_zstd` in crate `ffi`
   --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:791:38
    |
791 |     pub const ZSTD: Self = Self(ffi::TLSEXT_cert_compression_zstd as u16);
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `TLSEXT_cert_compression_zlib`
    |
   ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:217875
    |
  3 | ...i32 = 6 ; pub const TLSEXT_cert_compression_zlib : i32 = 1 ; pub const TLSEXT_cert_compression_brotli : i32 = 2 ; pub const TLSEXT_MAX...
    |              -------------------------------------------- similarly named constant `TLSEXT_cert_compression_zlib` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `RAMA_SSL_CTX_set_raw_cipher_list` in crate `ffi`
    --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:1429:22
     |
1429 |             cvt(ffi::RAMA_SSL_CTX_set_raw_cipher_list(
     |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `SSL_CTX_set_cipher_list`
     |
    ::: /build/source/codex-rs/target/x86_64-unknown-linux-gnu/release/build/rama-boring-sys-d706dd8e7e9f0af3/out/bindings.rs:3:827122
     |
   3 | ... { pub fn SSL_CTX_set_cipher_list (ctx : * mut SSL_CTX , str_ : * const :: std :: os :: raw :: c_char) -> :: std :: os :: raw :: c_int ; } ...
     |       ------------------------------------------------------------------------------------------------------------------------------------- similarly named function `SSL_CTX_set_cipher_list` defined here

error[E0425]: cannot find function, tuple struct or tuple variant `RAMA_SSL_CTX_set_extension_order` in crate `ffi`
    --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:1939:22
     |
1939 |             cvt(ffi::RAMA_SSL_CTX_set_extension_order(
     |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0425]: cannot find function, tuple struct or tuple variant `SSL_set_record_size_limit` in crate `ffi`
    --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:2985:27
     |
2985 |         unsafe { cvt(ffi::SSL_set_record_size_limit(self.as_ptr(), value) as c_int).map(|_| ()) }
     |                           ^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

error[E0425]: cannot find function, tuple struct or tuple variant `SSL_set_delegated_credential_schemes` in crate `ffi`
    --> /build/codex-0.91.0-vendor/rama-boring-0.5.9/src/ssl/mod.rs:2994:25
     |
2994 |             cvt_0i(ffi::SSL_set_delegated_credential_schemes(
     |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not found in `ffi`

   Compiling zstd v0.13.3
   Compiling codex-client v0.91.0 (/build/source/codex-rs/codex-client)
   Compiling codex-api v0.91.0 (/build/source/codex-rs/codex-api)
Some errors have detailed explanations: E0425, E0531.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `rama-boring` (lib) due to 16 previous errors
warning: build failed, waiting for other jobs to finish...
```

--- a/pkgs/by-name/co/codex/package.nix
+++ b/pkgs/by-name/co/codex/package.nix
@@ -4,6 +4,10 @@
   rustPlatform,
   fetchFromGitHub,
   installShellFiles,
+  clang,
+  cmake,
+  gitMinimal,
+  libclang,
   makeBinaryWrapper,
   nix-update-script,
   pkg-config,
@@ -14,26 +18,48 @@
 }:
 rustPlatform.buildRustPackage (finalAttrs: {
   pname = "codex";
-  version = "0.89.0";
+  version = "0.92.0";
 
   src = fetchFromGitHub {
     owner = "openai";
     repo = "codex";
     tag = "rust-v${finalAttrs.version}";
-    hash = "sha256-VFbtxGOqX80qWqVo+BG+BnUr8DiLCfcJCrN9fwy7utY=";
+    hash = "sha256-m/g+5wdehyaHDw6i5vik4HXiisY/iWFtPX0gKjCFPNY=";
   };
 
   sourceRoot = "${finalAttrs.src.name}/codex-rs";
 
-  cargoHash = "sha256-gg7KPEMO2aiBcIN8TllaDQeTLyw+WLfmMrXBKV/L53M=";
+  cargoHash = "sha256-fuT8vPb9/7fZam129nR6y+r+3j46WBhlf73Htkcjpzc=";
 
   nativeBuildInputs = [
+    clang
+    cmake
+    gitMinimal
     installShellFiles
     makeBinaryWrapper
     pkg-config
   ];
 
-  buildInputs = [ openssl ];
+  buildInputs = [
+    libclang
+    openssl
+  ];
+
+  # NOTE: set LIBCLANG_PATH so bindgen can locate libclang, and adjust
+  # warning-as-error flags to avoid known false positives (GCC's
+  # stringop-overflow in BoringSSL's a_bitstr.cc) while keeping Clang's
+  # character-conversion warning-as-error disabled.
+  env = {
+    LIBCLANG_PATH = "${lib.getLib libclang}/lib";
+    NIX_CFLAGS_COMPILE = toString (
+      lib.optionals stdenv.cc.isGNU [
+        "-Wno-error=stringop-overflow"
+      ]
+      ++ lib.optionals stdenv.cc.isClang [
+        "-Wno-error=character-conversion"
+      ]
+    );
+  };
 
   # NOTE: part of the test suite requires access to networking, local shells,
   # apple system configuration, etc. since this is a very fast moving target

